content/en/blog/2023/traffic-for-ambient-and-sidecar/index.md

```diff
@@ -0,0 +1,105 @@
+keywords: [traffic,ambient,sidecar,coexistence]
+---
+
+There are 2 modes for Istio: ambient and sidecar, the former is still on the way, the latter is the classic one. Therefore, the coexistence of ambient and sidecar should be a normal deployment form. That's the reason why this blog may be helpful for Istio users.
+
+## Background
+
+In the architecture of modern microservices, communication and management among services is critical. To address the challenge, Istio emerged as a service mesh technology. It provides traffic control, security, and superior observation capabilities by importing the sidecar. In order to further improve the adaptability and flexibility of Istio, the Istio community began to explore a new mode - ambient mode. In this mode, Istio no longer relies on explicit sidecar injection, but achieves communication and mesh management among services through ztunnel and waypoint proxies. Ambient also brings a series of improvements, such as lower resource consumption, simpler deployment, more flexible configuration options, and non-restarting for pods when enabling ambient, enabling Istio to play a better role in various scenarios.
+
+There are many blogs to introduce and analyze ambient in community and technology forums, and this blog will analyze the network traffic path in Istio ambient and sidecar modes. We will analyze the network traffic path between services in these two modes.
+
+In order to clarify the network traffic paths, this blog will explore two concrete scenarios with corresponding diagram to make it easier to understand. These two scenarios include:
+
+- **The network path of services in ambient mode to services in sidecar mode**
+- **The network path of services in sidecar mode to services in ambient mode**
+
+_Note 1: The following analysis is based on Istio 1.18.2, where ambient mode uses iptables for redirection._
+
+_Note 2: The communications between sidecar and ztunnel/waypoint proxy uses `[HTTP Based Overlay Network (HBONE)](https://docs.google.com/document/d/1Ofqtxqzk-c_wn0EgAXjaJXDHB9KhDuLe-W3YGG67Y8g/edit)`._
+
+## Ambient sleep to sidecar httpbin
+
+### Deployment and configuration for the first scenario
+
+- sleep is deployed in namespace foo
+    - sleep pod is scheduled to Node A
+- httpbin is deployed in namespace bar
+    - httpbin is scheduled to Node B
+- foo namespace enables ambient mode (foo namespace contains label: `istio.io/dataplane-mode=ambient`)
+- bar namespace enables sidecar injection (bar namespace contains label: `istio-injection: enabled`)
+
+With the above description, the deployment and network traffic paths are:
+
+{{< image width="100%"
+    link="ambient-to-sidecar.png"
+    caption="Ambient sleep to Sidecar httpbin"
+    >}}
+
+`ztunnel` will be deployed as a DaemonSet in istio-system namespace if ambient is enabled, while istio-cni and ztunnel would generate iptables rules and routes for both the ztunnel pod and pods on each node.
```

Comment:
```suggestion
ztunnel will be deployed as a DaemonSet in istio-system namespace if ambient is enabled, while istio-cni and ztunnel would generate iptables rules and routes for both the ztunnel pod and pods on each node.
```

For consistency, either backtick all or none. I don't believe ztunnel needs to be backticked and that seems to be the primary method employed throughout the blog.

Reply:
sure, already done!